src/main/scala/com/expedia/www/haystack/commons/config/ConfigurationLoader.scala

```diff
@@ -14,70 +14,88 @@
  *     limitations under the License.
  *
  */
+
 package com.expedia.www.haystack.commons.config
 
 import java.io.File
 
-import com.typesafe.config.{Config, ConfigFactory, ConfigRenderOptions}
-import org.apache.commons.codec.binary.StringUtils
+import com.typesafe.config.{Config, ConfigFactory, ConfigRenderOptions, ConfigValueType}
 import org.slf4j.LoggerFactory
 
 import scala.collection.JavaConverters._
 
 object ConfigurationLoader {
 
-  val ENV_NAME_PREFIX = "HAYSTACK_PROP_"
-
   private val LOGGER = LoggerFactory.getLogger(ConfigurationLoader.getClass)
 
+  private[haystack] val ENV_NAME_PREFIX = "HAYSTACK_PROP_"
+
   /**
-    * Loads a HOCON config file and overrides entries in them with values, if specified, from environment variables
-    * with given prefix
-    * For example, if the envNamePrefix given is HAYSTACK_ and if an environment variable exist with the name
-    * HAYSTACK_KAFKA_STREAMS_NUM_STREAM_THREADS, then a config entry with key kafka.streams.num.stream.threads
-    * will be added to the config object or an existing value in the config object will be overwritten
+    * Load and return the configuration
+    * if overrides_config_path env variable exists, then we load that config file and use base conf as fallback,
+    * else we load the config from env variables(prefixed with haystack) and use base conf as fallback
     *
-    * @param resourceName name of the resource file to be loaded. Default value is `config/base.conf`
-    * @param envNamePrefix env variable prefix to override config values. Default is `HAYSTACK_PROP_`
-    *
-    * @return an instance of com.typesafe.Config
     */
-  def loadConfigFileWithEnvOverrides(resourceName : String = "config/base.conf", envNamePrefix : String = ENV_NAME_PREFIX) : Config = {
-
-    require( envNamePrefix != null && envNamePrefix.length > 0 , "envNamePrefix is required")
-    require( resourceName != null && resourceName.length > 0 , "resourceName is required")
+  lazy val loadAppConfig: Config = {
```

Comment:
@ashishagg I purposely avoided doing this with the signature and comments. I made the filename and env prefix as default -so `loadAppConfig` and `loadConfigFileWithEnvOverrides` do the same without any arguments. But it doesnt restrict consumers to just config/base.conf